### ISSUE TITLE
React cells are re-rendered when loading a new page via the InfiniteRowModel

### ISSUE THREAD

**Issue by HerbiePorter** [ag-grid/ag-grid#4388]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**
When a new page is loaded via the InfiniteRowModel all visible cells using custom React renderers will re-render.  This causes the whole grid to flash if you're using lots of custom cells.

This is reproducible here: https://plnkr.co/edit/lvjUkjNmeOAEDdWL

When you scroll to the bottom you can see all the visible cells in the Athlete column with the custom React renderer flicker and the render count increase by 1.

**Expected behavior**
Loading a new page via the InfiniteRowModel should not cause the visible cells to re-render.

**Please tell us about your environment:**

* **AG Grid version:** X.X.X
v24.0.0 and v25.0.0

* **Browser:**
Mozilla/5.0 (Macintosh; Intel Mac OS X 11_2_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.72 Safari/537.36

* **Language:** 
TypeScript/ES6

**Comment by makinggoodsoftware** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by kiril-matev** [comment]
Hello @HerbiePorter!

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference and description:
**AG-5222 When using infinite row model, displayed cell values get rendered a second time when a new block of data is loaded**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

You can see the next release date and whether this item is selected for implementation on the NEXT RELEASE tab of the product pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

**Please note we will not update this GitHub issue when this bug is fixed in a future version of AG Grid.**

**The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:**
https://www.ag-grid.com/ag-grid-changelog/

Thanks again for bringing this up with us - we appreciate your time!

**Comment by makinggoodsoftware** [comment]
Hi,

In order to help us keep our issues queue clear, we are going to close this issue

The official way to check the status for an issue raised here is to check its AG-xxxx ID against our pipeline and changelog

…the pipeline (if not yet developed)

https://www.ag-grid.com/ag-grid-pipeline/

... or the changelog, (if the issue has been resolved)

https://www.ag-grid.com/ag-grid-changelog/

We keep issues raised in the pipeline through github open for a while so that the community can add some feedback to them, and close them when there is no activity for a while.

If you want to add further feedback to a closed ticket with this comment, we would recommend opening a new ticket as we don’t monitor closed issues.

**Comment by HerbiePorter** [comment]
@makinggoodsoftware I can't see issue AG-5222 in the pipeline or in the backlog anymore.  Do you know what happened to it?

**Comment by kiril-matev** [comment]
Hi @HerbiePorter!

Thanks for following up on this.

AG-5222 was resolved in our latest release - v25.2. This is why it's no longer appearing in the pipeline.

Instead as it's been implemented, it needs to appear in the changelog here:
https://www.ag-grid.com/ag-grid-changelog/

However, sometimes issues don't get picked up correctly by the changelog and it takes a couple of hours for them to be refreshed. We will look into this so issues appear correctly.

In any case, thanks for reaching out. Please resolve this issue by upgrading to v25.2.
